### ISSUE TITLE
OpenEyeToolkitWrapper.calculate_partial_charges_am1bcc should append OE error to thrown error 

### ISSUE THREAD

**Issue by j-wags** [openforcefield/openforcefield#379]
_Based on question from @yudongqiu_

When OETKW charge generation fails, the toolkit throws an error that simply says "Unable to assign charges". OE has usually printed some sort of warning beforehand, but that is on some other output channel (STDOUT?) and so users need to go rooting around potentially-verbose output to find it. We should see if we can capture this output and append it to the error message.

Relevant code is here: https://github.com/openforcefield/openforcefield/blob/5c21a072f5dce44ec2f9c35cbeb0077edddbf0a9/openforcefield/utils/toolkits.py#L1255-L1258

**Comment by davidlmobley** [comment]
Agree, yes. I think this is an issue sometimes more generally beyond just assign charges.

I believe there is a way to adjust the OE error/warning level to make sure things get handled better.

**Comment by j-wags** [comment]
@trevorgokey Here's how to catch the thrown error: https://github.com/openforcefield/openforcefield/issues/346#issuecomment-505202862

**Comment by dotsdl** [comment]
Hey @trevorgokey, I committed to taking this issue on along with #305 and #381. Is there any progress you'd like to report to help me get started, or is this perhaps complete?

**Comment by dotsdl** [comment]
Do we have an MWE to test a potential fix? I have tried the one in #346, but I'm getting another issue:

```python
from openforcefield.topology import Molecule
from openforcefield.topology import Topology
from openforcefield.typing.engines.smirnoff import ForceField

from openeye import oechem

oemol = oechem.OEMol()

with oechem.oemolistream("2799.oeb") as ifs:
    oechem.OEReadMolecule(ifs, oemol)

# This is a temporary bug fix for the bug #344 
#oechem.OE3DToAtomStereo(oemol)
#oechem.OE3DToBondStereo(oemol)

molecule = Molecule(oemol)

forcefield = ForceField('test_forcefields/smirnoff99Frosst.offxml')

topology = Topology.from_molecules(molecule)
system = forcefield.create_openmm_system(topology)
```
Gives:
```
---------------------------------------------------------------------
AttributeError                      Traceback (most recent call last)
<ipython-input-16-8eb74b3b097e> in <module>
     19 
     20 topology = Topology.from_molecules(molecule)
---> 21 system = forcefield.create_openmm_system(topology)

~/Projects/threads/molmat/openforcefield/issue-379/openforcefield/openforcefield/typing/engines/smirnoff/forcefield.py in create_openmm_system(self, topology, **kwargs)
   1102         """
   1103         # Make a deep copy of the topology so we don't accidentally modify it
-> 1104         topology = copy.deepcopy(topology)
   1105 
   1106         # Set the topology aromaticity model to that used by the current forcefield

~/.conda/envs/off-379/lib/python3.7/copy.py in deepcopy(x, memo, _nil)
    178                     y = x
    179                 else:
--> 180                     y = _reconstruct(x, memo, *rv)
    181 
    182     # If is its own copy, don't memoize.

~/.conda/envs/off-379/lib/python3.7/copy.py in _reconstruct(x, memo, func, args, state, listiter, dictiter, deepcopy)
    278     if state is not None:
    279         if deep:
--> 280             state = deepcopy(state, memo)
    281         if hasattr(y, '__setstate__'):
    282             y.__setstate__(state)

~/.conda/envs/off-379/lib/python3.7/copy.py in deepcopy(x, memo, _nil)
    148     copier = _deepcopy_dispatch.get(cls)
    149     if copier:
--> 150         y = copier(x, memo)
    151     else:
    152         try:

~/.conda/envs/off-379/lib/python3.7/copy.py in _deepcopy_dict(x, memo, deepcopy)
    238     memo[id(x)] = y
    239     for key, value in x.items():
--> 240         y[deepcopy(key, memo)] = deepcopy(value, memo)
    241     return y
    242 d[dict] = _deepcopy_dict

~/.conda/envs/off-379/lib/python3.7/copy.py in deepcopy(x, memo, _nil)
    178                     y = x
    179                 else:
--> 180                     y = _reconstruct(x, memo, *rv)
    181 
    182     # If is its own copy, don't memoize.

~/.conda/envs/off-379/lib/python3.7/copy.py in _reconstruct(x, memo, func, args, state, listiter, dictiter, deepcopy)
    305                 key = deepcopy(key, memo)
    306                 value = deepcopy(value, memo)
--> 307                 y[key] = value
    308         else:
    309             for key, value in dictiter:

~/Projects/threads/molmat/openforcefield/issue-379/openforcefield/openforcefield/topology/molecule.py in __hash__(self)
   1648         string
   1649         """
-> 1650         return hash(self.to_smiles())
   1651 
   1652     @classmethod

~/Projects/threads/molmat/openforcefield/issue-379/openforcefield/openforcefield/topology/molecule.py in to_smiles(self, isomeric, explicit_hydrogens, mapped, toolkit_registry)
   1864         """
   1865         # Initialize cached_smiles dict for this molecule if none exists
-> 1866         if self._cached_smiles is None:
   1867             self._cached_smiles = {}
   1868 

AttributeError: 'FrozenMolecule' object has no attribute '_cached_smiles'
```
I believe since that issue is closed, however, that without this new error we might not have a good candidate for testing a solution to #379 here.

**Comment by j-wags** [comment]
Ah, yes. Perhaps ironically given your work on #305, one way to reproduce this error currently is to try and run AM1-BCC on a Lithium ion:
```
from openforcefield.topology import Molecule
from openforcefield.topology import Topology
from openforcefield.typing.engines.smirnoff import ForceField

from openeye import oechem

molecule = Molecule.from_smiles('[Li+1]')
forcefield = ForceField('test_forcefields/smirnoff99Frosst.offxml')

topology = Topology.from_molecules(molecule)
system = forcefield.create_openmm_system(topology)
```

**Comment by dotsdl** [comment]
Closing, as this is addressed in #573.
